packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -1085,6 +1125,47 @@ mixin SchedulerBinding on BindingBase {
     }
   }
 
+  final Map<Object, DartPerformanceMode> _performanceModes = <Object, DartPerformanceMode>{};
+
+  /// Request a specific [DartPerformanceMode].
+  ///
+  /// Requestor passes a handle (`requestor`) when making the request, this is typically
+  /// the component itself. Returns `null` if the request was not successful due to
+  /// conflicting performance mode requests. Requestor is responsible for calling
+  /// [PerformanceModeRequestHandle.dispose] when it no longer requires the performance mode.
+  PerformanceModeRequestHandle? createPerformanceModeRequest(Object requestor, DartPerformanceMode mode) {
```

Comment:
should this be called `requestPerformanceMode`? You're not only creating the request, you're actually making it.